pylzy/tests/api/v1/test_calls.py

```diff
@@ -0,0 +1,242 @@
+        func: FuncSignature = wf.owner.runtime.calls[0].signature.func
+        self.assertEqual(str, func.output_types[0])
+        self.assertEqual(1, len(func.input_types))
+        self.assertEqual(List[str], func.input_types['arg'])
+
+    def test_custom_class(self):
+        with self.lzy.workflow("test") as wf:
+            # noinspection PyTypeChecker
+            call_custom_class(A())
+
+        # noinspection PyUnresolvedReferences
+        func: FuncSignature = wf.owner.runtime.calls[0].signature.func
+        self.assertEqual(A, func.output_types[0])
+        self.assertEqual(1, len(func.input_types))
+        self.assertEqual(A, func.input_types['arg'])
+
+    def test_custom_class_invalid(self):
+        class B:
+            pass
+
+        with self.assertRaisesRegex(ValidationError, "instance of A expected"):
+            with self.lzy.workflow("test"):
+                # noinspection PyTypeChecker
+                call_custom_class(B())
+
+    def test_type_from_arg(self):
+        @op
+        def returns_list_str() -> List[str]:
+            pass
+
+        with self.lzy.workflow("test") as wf:
+            value = returns_list_str()
+            self.assertFalse(materialized(value))
+            call_no_arg_hint(value)
+
+        # noinspection PyUnresolvedReferences
+        func: FuncSignature = wf.owner.runtime.calls[1].signature.func
+        self.assertEqual(str, func.output_types[0])
+        self.assertEqual(1, len(func.input_types))
+        self.assertEqual(List[str], func.input_types['arg'])
+
+    def test_returns_none(self):
+        @op
+        def returns_none() -> None:
+            pass
+
+        with self.lzy.workflow("test"):
+            # noinspection PyNoneFunctionAssignment
+            result = returns_none()
+            self.assertIsNone(result)
+
+    def test_none_arguments(self):
+        with self.lzy.workflow("test") as wf:
+            call_no_arg_hint(None)
+
+        # noinspection PyUnresolvedReferences
+        func: FuncSignature = wf.owner.runtime.calls[0].signature.func
+        self.assertEqual(1, len(func.input_types))
+        self.assertEqual(type(None), func.input_types['arg'])
+
+    def test_varargs(self):
+        # noinspection PyUnusedLocal
+        @op
+        def varargs(c: float, *args, **kwargs) -> None:
+            pass
+
+        with self.lzy.workflow("test") as wf:
+            varargs(2.0, *(1, 2, 3), **{'a': "a", 'b': 2})
+
+        # noinspection PyUnresolvedReferences
+        func: FuncSignature = wf.owner.runtime.calls[0].signature.func
+        self.assertEqual(6, len(func.input_types))
+        self.assertEqual(str, func.input_types['a'])
+        self.assertEqual(int, func.input_types['b'])
+        self.assertEqual(float, func.input_types['c'])
+        self.assertEqual(('a', 'b'), func.kwarg_names)
+
+    def test_multiple_return_values(self):
+        # noinspection PyUnusedLocal
+        @op
+        def multiple_return() -> (int, str):
```

Comment:
Why some ops are defined here, and some are higher?

Reply:
If ops are used multiple times they defined globally